src/compile/data/assemble.ts

```diff
@@ -36,6 +42,79 @@ function removeUnnecessaryNodes(node: DataFlowNode) {
 
   node.children.forEach(removeUnnecessaryNodes);
 }
+
+/**
+ * Clones the subtree and ignores output nodes except for the leafs, which are renamed.
+ */
+function cloneSubtree(facet: FacetNode) {
+  function clone(node: DataFlowNode): DataFlowNode[] {
+    if (isClonable(node)) {
+      const copy = node.clone();
+
+      if (copy instanceof OutputNode) {
+        const newName = FACET_SCALE_PREFIX + facet.model.getName(copy.source);
+        copy.source = newName;
+
+        facet.model.component.data.outputNodes[newName] = copy;
+
+        flatten(node.children.map(clone)).forEach((n: DataFlowNode) => n.parent = copy);
+      } else if (copy instanceof AggregateNode) {
```

Comment:
Just like `AggregateNode`, you have to augment `groupby` of `StackNode` as well.